services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/asn1/SpatDecoder.java

```diff
@@ -0,0 +1,175 @@
+import us.dot.its.jpo.geojsonconverter.validator.SpatJsonValidator;
+import us.dot.its.jpo.ode.api.models.messages.DecodedMessage;
+import us.dot.its.jpo.ode.api.models.messages.EncodedMessage;
+import us.dot.its.jpo.ode.context.AppContext;
+import us.dot.its.jpo.ode.model.Asn1Encoding;
+import us.dot.its.jpo.ode.model.Asn1Encoding.EncodingRule;
+import us.dot.its.jpo.ode.model.OdeAsn1Data;
+import us.dot.its.jpo.ode.model.OdeAsn1Payload;
+import us.dot.its.jpo.ode.model.OdeSpatData;
+import us.dot.its.jpo.ode.model.OdeSpatMetadata;
+import us.dot.its.jpo.ode.model.OdeSpatMetadata.SpatSource;
+import us.dot.its.jpo.ode.model.OdeSpatPayload;
+import us.dot.its.jpo.ode.model.OdeData;
+import us.dot.its.jpo.ode.model.OdeHexByteArray;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.RecordType;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.SecurityResultCode;
+import us.dot.its.jpo.ode.model.OdeLogMsgMetadataLocation;
+import us.dot.its.jpo.ode.model.OdeMsgPayload;
+import us.dot.its.jpo.ode.model.ReceivedMessageDetails;
+import us.dot.its.jpo.ode.model.RxSource;
+import us.dot.its.jpo.ode.plugin.j2735.J2735IntersectionState;
+import us.dot.its.jpo.ode.plugin.j2735.builders.SPATBuilder;
+import us.dot.its.jpo.ode.util.JsonUtils;
+import us.dot.its.jpo.ode.util.XmlUtils;
+import us.dot.its.jpo.ode.util.XmlUtils.XmlUtilsException;
+
+
+@Component
+public class SpatDecoder implements Decoder {
+
+    @Autowired
+    SpatJsonValidator spatJsonValidator;
+
+    public SpatProcessedJsonConverter converter = new SpatProcessedJsonConverter();
+
+
+    @Override
+    public DecodedMessage decode(EncodedMessage message) {
+        
+        // Convert to Ode Data type and Add Metadata
+        OdeData data = getAsOdeData(message.getAsn1Message());
+
+        XmlUtils xmlUtils = new XmlUtils();
+
+        try {
+            // Convert to XML for ASN.1 Decoder
+            String xml = xmlUtils.toXml(data);
+
+            // Send String through ASN.1 Decoder to get Decoded XML Data
+            String decodedXml = DecoderManager.decodeXmlWithAcm(xml);
+            
+
+            // Convert to Ode Json 
+            OdeSpatData spat = getAsOdeJson(decodedXml);
+
+            // build output data structure
+            try{
+                ProcessedSpat processedSpat = createProcessedSpat(spat);
+                return new SpatDecodedMessage(processedSpat, spat, message.getAsn1Message(), "");
+            } catch(Exception e) {
+                return new SpatDecodedMessage(null, spat, message.getAsn1Message(), e.getMessage());
+            }
+            
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return new SpatDecodedMessage(null, null, message.getAsn1Message(), e.getMessage());
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new SpatDecodedMessage(null, null, message.getAsn1Message(), e.getMessage());
+        }
```

Comment:
chore(blocking): use proper log statement here instead of calling e.printStackTrace()

suggestion(non-blocking): combine into one catch statement since both are doing the same thing

```suggestion
        } catch (Exception e) {
            log.error("Failed to decode SPaT: {}", e.getMessage());
            return new SpatDecodedMessage(null, null, message.getAsn1Message(), e.getMessage());
        }
```